compiler/ast/src/main/kotlin/motif/ast/compiler/CompilerType.kt

```diff
@@ -81,9 +81,12 @@ class CompilerType(
             return type
         }
 
-        val superType = MoreTypes.nonObjectSuperclass(env.typeUtils, env.elementUtils, type).orNull() ?: return null
+        env.typeUtils.directSupertypes(type).forEach { superType ->
+            getMatchingSuperType(baseType, superType as DeclaredType)?.let { return it }
```

Comment:
Can this cast fail?

Reply:
Yes for error types but that assumption is made across the codebase. To handle error types correctly a more holistic approach is needed.

Reply:
Link a followup issue?

Reply:
#51